feature(update): improve consistency of updates to saves

SpenserJ opened the pull request:
This fixes a bug that could occur if you import and old save, but refresh between the save being updated, and the save being persisted. This will also prevent the game from loading if an update fails, and will provide additional details about which update failed and the state before and after the failure:
![image](https://user-images.githubusercontent.com/90011/96664680-8e231380-1310-11eb-97a7-19b1c4087e5d.png)

RedSparr0w reviewed (changes requested):
Changes look good to me!

Could we also add a download backup save button to the error message so it will be easier for us to debug if we do encounter any problems.
![image](https://user-images.githubusercontent.com/7288322/96690567-54b9d900-13e0-11eb-93d4-fde38696143f.png)

Edit:
Another thing I just realized, you can't access the start menu if your save has an error, which means you are stuck in the loading screen.

RedSparr0w commented on src/scripts/Update.ts:
```diff
@@ -53,311 +239,86 @@ class Update implements Saveable {
+                        title: `Failed to update to v${this.version}!`,
+                        message: 'Please check the console for errors, and report them on our Discord.',
+                        type: NotificationConstants.NotificationOption.danger,
+                        timeout: 6e4,
```
Maybe we should extend the timeout of this notification.
```suggestion
                        timeout: GameConstants.HOUR,
```

SpenserJ replied:
I set it to a full day, just in case someone walks away from the computer mid-update. If they take longer, the notification will appear, but most people will refresh the page and get the error again

SpenserJ commented:
> Could we also add a download backup save button to the error message so it will be easier for us to debug if we do encounter any problems.
> Another thing I just realized, you can't access the start menu if your save has an error, which means you are stuck in the loading screen.

I've added a download button, changed the colors (because it was red on red), and added a reset button to the save because the start menu would have needed to be reworked so most of the items were disabled, and the z-index would need to be lifted for the menu and settings modals.
![image](https://user-images.githubusercontent.com/90011/96729922-e7239380-1372-11eb-8a19-ac6d01a7aba4.png)
